river_dl/preproc_utils.py

```diff
@@ -19,13 +45,25 @@ def asRunConfig(config, code_dir, outFile):
         branch = ref.split("/")[-1]
     with open(os.path.join(code_dir,'.git/', ref),'r') as git_hash:
         commit = git_hash.readline().strip()
+    status = str(subprocess.Popen(['git status'], shell=True,stdout=subprocess.PIPE).communicate()[0]).split("\\n")
```

Comment:
Could you change this so it doesn't assume you are in the git repo?

I think this would work:
```suggestion
    status = str(subprocess.Popen([f'git -C {code_dir} status'], shell=True,stdout=subprocess.PIPE).communicate()[0]).split("\\n")
```

Reply:
Thanks. I had tried specifying the cwd for popen before, but that (and this suggestion) had issues when run from the upper directory of the repo. This seems to work:

`status = str(subprocess.Popen(['git status'], cwd = None if code_dir=="" else code_dir,shell=True,stdout=subprocess.PIPE).communicate()[0]).split("\\n")`